src/include/ikarus/finiteElements/mechanics/enhancedAssumedStrains.hh

```diff
@@ -0,0 +1,406 @@
+    const auto& referenceElement = Dune::ReferenceElements<double,2>::general(geometry.type());
+    const auto quadPos0 = referenceElement.position(0,0);
+
+    const auto jacobianinvT0
+        = toEigenMatrix(geometry.jacobianInverseTransposed(quadPos0));  // J^{-1}.Transpose() in Dune = J^{-1}
+    const auto detJ0 = geometry.integrationElement(quadPos0);           // determinant(J)
+
+    auto jaco = (jacobianinvT0).inverse().eval();
+    auto J11  = jaco(0, 0);
+    auto J12  = jaco(0, 1);
+    auto J21  = jaco(1, 0);
+    auto J22  = jaco(1, 1);
+
+    Eigen::Matrix3d T0;
+    T0 << J11 * J11, J12 * J12, J11 * J12, J21 * J21, J22 * J22, J21 * J22, 2.0 * J11 * J21, 2.0 * J12 * J22,
+        J21 * J12 + J11 * J22;
+
+    return T0.inverse() * detJ0;
+  }
+
+  template <typename Geometry>
+  Eigen::Matrix<double, 6, 6> calcTransformationMatrix3D(const Geometry& geometry) {
+    const auto& referenceElement = Dune::ReferenceElements<double,3>::general(geometry.type());
+    const auto quadPos0 = referenceElement.position(0,0);
+
+    const auto jacobianinvT0
+        = toEigenMatrix(geometry.jacobianInverseTransposed(quadPos0));  // J^{-1}.Transpose() in Dune = J^{-1}
+    const auto detJ0 = geometry.integrationElement(quadPos0);           // determinant(J)
+
+    auto jaco = (jacobianinvT0).inverse().eval();
+    auto J11  = jaco(0, 0);
+    auto J12  = jaco(0, 1);
+    auto J13  = jaco(0, 2);
+    auto J21  = jaco(1, 0);
+    auto J22  = jaco(1, 1);
+    auto J23  = jaco(1, 2);
+    auto J31  = jaco(2, 0);
+    auto J32  = jaco(2, 1);
+    auto J33  = jaco(2, 2);
+
+    Eigen::Matrix<double, 6, 6> T0;
+    T0 << J11 * J11, J12 * J12, J13 * J13, J11 * J12, J11 * J13, J12 * J13, J21 * J21, J22 * J22, J23 * J23, J21 * J22,
+        J21 * J23, J22 * J23, J31 * J31, J32 * J32, J33 * J33, J31 * J32, J31 * J33, J32 * J33, 2.0 * J11 * J21,
+        2.0 * J12 * J22, 2.0 * J13 * J23, J11 * J22 + J21 * J12, J11 * J23 + J21 * J13, J12 * J23 + J22 * J13,
+        2.0 * J11 * J31, 2.0 * J12 * J32, 2.0 * J13 * J33, J11 * J32 + J31 * J12, J11 * J33 + J31 * J13,
+        J12 * J33 + J32 * J13, 2.0 * J31 * J21, 2.0 * J32 * J22, 2.0 * J33 * J23, J31 * J22 + J21 * J32,
+        J31 * J23 + J21 * J33, J32 * J23 + J22 * J33;
+
+    return T0.inverse() * detJ0;
+  }
+
+  template <typename Geometry>
+  struct EASQ1E4 {
+    static constexpr int strainSize         = 3;
+    static constexpr int enhancedStrainSize = 4;
+
+    EASQ1E4() = default;
+    explicit EASQ1E4(const Geometry& geometry)
+        : geometry{std::make_unique<Geometry>(geometry)}, T0InverseTransformed{calcTransformationMatrix2D(geometry)} {}
+
+    auto calcM(const Dune::FieldVector<double, 2>& quadPos) const {
+      Eigen::Matrix<double, strainSize, enhancedStrainSize> M;
+      M.setZero(strainSize, enhancedStrainSize);
+      const double xi   = quadPos[0];
+      const double eta  = quadPos[1];
+      M(0, 0)           = xi - 0.5;
```

Comment:
this should be further investigated. In Maple i have a file where i derived. There should be:

2*xi-1 instead and similar for the other monomials.

Reply:
![image](https://user-images.githubusercontent.com/10537464/193272009-0d8aced7-32d1-478f-9ab2-cd532a4975a0.png)